futurecar/src/components/Store/Store.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import styled from 'styled-components';
+import axios from 'axios';
+const baseURL = "https://us-central1-future-apis.cloudfunctions.net/futureCar"
+
+const CarrosGrid = styled.div `
+    max-width: 650px;
+`
+
+const DivCarrosStyle = styled.div`
+display: flex;
+flex-wrap: wrap;
+margin: auto 0;
+
+`
+const InputDivCarros = styled.div`
+display:block;
+margin-bottom: 20px;
+`
+
+const CardCar = styled.div `
+    display: block;
+    margin: 10px;
+    width: 290px;
+    border: 1px dotted black;
+`;
+
+const ImagemCar = styled.img `
+    width: 100%;
+`;
+
+class Store extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            todosCarros: [],
+            menor: '',
+            maior: Infinity
+            }
+        }
+    
+    pegarCarros = async () => {
+        const response = await axios.get(`${baseURL}/cars`)
+        this.setState({todosCarros: response.data.cars})
+        console.log(response.data.cars)
+    }
+    componentDidMount(){
+        this.pegarCarros();
+    }
+
+    // controleMenor = event => {
+    //     this.setState({menor: event.target.value})
+    // }
+
+    // controleMaior = event => {
+    //     this.setState({maior: event.target.value})
+    // }
+
+    
+       
+
+    render() {
+
+        let filtroCarro = this.state.todosCarros.filter((carro)=>{
+            return Number(carro.price) < (this.state.maior || Infinity)
```

Comment:
Aqui só funcionou porque o Javascript converte `string` em `number` na hora de fazer comparações de maior e menor!
O `this.state.maior` e o `this.state.menor` são `string` e não `number`! Então, vocês deveriam ter feito assim (converter os dois para número):
`return Number(carro.price) < Number(this.state.maior || Infinity)`